pkg/trait/logging.go

```diff
@@ -80,6 +91,30 @@ func (l loggingTrait) Apply(e *Environment) error {
 			envvar.SetVal(&e.EnvVars, envVarQuarkusConsoleColor, True)
 		}
 	}
+}
 
-	return nil
+func (l loggingTrait) setCatalogConfiguration(e *Environment) {
+	if e.ApplicationProperties == nil {
+		e.ApplicationProperties = make(map[string]string)
+	}
+	e.ApplicationProperties["camel.k.logging.level"] = l.Level
```

Comment:
not for this first iteration but we could leverage tags, i.e:

```go
type Logging struct {
	Level string `property:"camel.k.logging.level"`
}
```

So the mapping trait -> properties could be made generic and most of the time, not manually hadn crafted.

Reply:
Yeah. However the problem there could be how it affects the API. IIUC the conversion would happen when translating the CamelCatalog into a CR. We had included the trait configuration some time ago, making the CRD longer than probably should be. I wonder if adding more stuff may reach some limit that can put us in troubles. In any case, some future challenge we will tackle.

Reply:
I don't think I follow, what I mean is that instead of doing this:

```go
    e.ApplicationProperties["camel.k.logging.level"] = l.Level
```

We can leverage the tags defined on the go struct:

```go
    e.ApplicationProperties[GetProperty(l.Level)] = l.Level
```

Which can be further improved with reflection.
It should only impact the go code and eventually the documentation but not the CRDs.

Reply:
but this can be deferred to a later stage